core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -48,7 +48,7 @@
  * <p>Can be constructed manually but will often be generated by stub code generators.
  */
 @Immutable
-public class MethodDescriptor<ReqT, RespT> {
+public final class MethodDescriptor<ReqT, RespT> {
 
   private final MethodType type;
```

Comment:
Yes, the extra finals are not needed. I secretly suspect that the final added to the class might need some transition period internally, so I don't want to lose the finals  in that case.  I would be happy to remove them once this is merged internally.